util/src/main/java/org/datacommons/util/LogWrapper.java

```diff
@@ -49,10 +51,13 @@
   public final boolean persistLog;
   private Debug.Log.Builder log;
 
-  private Instant lastStatusAt;
+  // Update with lock, but read without object lock.
+  private volatile Instant lastStatusAt;
+  // Updated and read with object lock.
   private long countAtLastStatus = 0;
-  private long currentCount = 0;
-  private final Set<String> countersWithErrors = new HashSet<>();
+  private AtomicLong currentCount = new AtomicLong();
+  // A copy of the counters that are updated into the Log before persisting.
+  private final List<ConcurrentHashMap<String, Long>> counterMaps = new ArrayList<>();
```

Comment:
nit: add comment that position in list corresponds to number for Debug.Log.Level of the counterMap (that's what I inferred from the rest of this file, but not sure if I'm right)

Reply:
Done